library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2677,32 +2726,32 @@ impl<A: PartialEq> PartialEq for VecDeque<A> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<A: Eq> Eq for VecDeque<A> {}
+impl<T: Eq, A: Allocator> Eq for VecDeque<T, A> {}
 
-__impl_slice_eq1! { [] VecDeque<A>, Vec<B>, }
-__impl_slice_eq1! { [] VecDeque<A>, &[B], }
-__impl_slice_eq1! { [] VecDeque<A>, &mut [B], }
-__impl_slice_eq1! { [const N: usize] VecDeque<A>, [B; N], }
-__impl_slice_eq1! { [const N: usize] VecDeque<A>, &[B; N], }
-__impl_slice_eq1! { [const N: usize] VecDeque<A>, &mut [B; N], }
+__impl_slice_eq1! { [] VecDeque<T, A>, Vec<U, A>, }
```

Comment:
Hmm, why do you require both to come from the same allocator? Just comparing them shouldn't do any reallocation, right?

Reply:
I agree that it is not required, but I chose to copy [what `Vec` does](https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html#impl-PartialEq%3CVec%3CU%2C%20A%3E%3E)

Reply:
For all function potentially mixing allocators, while I agree that allowing it would be great, current APIs (eg for `Vec`) do not seem to go in this direction.

There may be a reason I don't know for this choice, but I chose consistency anyway. If the opposite is decided, I'll be happy to change this.